### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by ZanCorDX**
## 📝 Summary

High-level doc about LiveBuilder.

## 💡 Motivation and Context

- It was really hard to understand the whole system from scratch.
- I'm so nice that I love making docs for the comunity.
- I had a fight with my wife and I needed an excuse to stay a few more hours at the office.

## ✅ I have completed the following steps:

* [ ] Run `make lint`
* [ ] Run `make test`
* [ ] Added tests (if applicable)

### CODE REVIEW

File: docs/RBUILDER_CLASSES_AND_DATAFLOW.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
+# rbuilder Classes and Dataflow
```
Comment (jakubhruby7) [inline docs/RBUILDER_CLASSES_AND_DATAFLOW.md]: I'd change this to something like `LiveBuilder Dataflow` and change the file name too.

File: docs/RBUILDER_CLASSES_AND_DATAFLOW.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+The [`LiveBuilder`](../crates/rbuilder/src/live_builder/mod.rs) struct is the main component of rbuilder.
+
+## Core Components
+
+To create a `LiveBuilder`, you need the following core components:
```
Comment (jakubhruby7) [inline docs/RBUILDER_CLASSES_AND_DATAFLOW.md]: a `LiveBuilder` instance, ...

File: docs/RBUILDER_CLASSES_AND_DATAFLOW.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    - LiveBuilder::sink_factory
+    - LiveBuilder::builders
+  - Triggers block building tasks
+
+- **payload_events_channel**:
+  - A channel of [`MevBoostSlotData`](../crates/rbuilder/src/live_builder/payload_events/mod.rs)
+  - Receives block-building opportunities
+  - Each received `MevBoostSlotData` triggers a new block-building task via `BlockBuildingPool`
+  - Sources slots from `LiveBuilder::blocks_source`
+
```
Comment (jakubhruby7) [inline docs/RBUILDER_CLASSES_AND_DATAFLOW.md]: we can add a section `Dataflow Diagram` as this diagram refers to two previous sections

File: docs/RBUILDER_CLASSES_AND_DATAFLOW.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
+# rbuilder Classes and Dataflow
+
+The [`LiveBuilder`](../crates/rbuilder/src/live_builder/mod.rs) struct is the main component of rbuilder.
+
+## Core Components
+
+To create a `LiveBuilder`, you need the following core components:
+
+1. `blocks_source`: The source of slots to build. Implements the [`SlotSource`](../crates/rbuilder/src/live_builder/mod.rs) trait. This abstraction enables rbuilder to handle block building in various contexts:
```
Comment (jakubhruby7) [inline docs/RBUILDER_CLASSES_AND_DATAFLOW.md]: let's link `block_source`

File: docs/RBUILDER_CLASSES_AND_DATAFLOW.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+   - L1: Consensus client generating slots with potential forks
+   - L2: Sequencer generating slots
+
+2. `builders`: A vector of objects implementing the [`BlockBuildingAlgorithm`](../crates/rbuilder/src/building/builders/mod.rs) trait. Each builder:
```
Comment (jakubhruby7) [inline docs/RBUILDER_CLASSES_AND_DATAFLOW.md]: let's link `builders`

File: docs/RBUILDER_CLASSES_AND_DATAFLOW.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+   - Takes a base block state and a stream of simulated orders
+   - Continuously generates new blocks
+   - Optimizes to maximize the true block value
+
+3. `sink_factory`: A factory for the destination of built blocks. Implements [`UnfinishedBlockBuildingSinkFactory`](../crates/rbuilder/src/building/builders/mod.rs). This abstraction supports different contexts:
```
Comment (jakubhruby7) [inline docs/RBUILDER_CLASSES_AND_DATAFLOW.md]: let's link `sink_factory`

File: docs/RBUILDER_CLASSES_AND_DATAFLOW.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+   - L1: Requires bidding
+   - L2: No bidding needed
+   - Testing environments
+
+## Initialization Process
+
+The main entrypoint `LiveBuilder::run()` initializes several long-lived components:
+
+- **RPC Module**:
```
Comment (jakubhruby7) [inline docs/RBUILDER_CLASSES_AND_DATAFLOW.md]: can we link to the RPC Module here?

**Review by jakubhruby7 — COMMENTED** [review]
Good doc! Left a few minor comments. Let's make sure we link to the code when we refer a specific struct/trait etc.

**Review by ferranbt — APPROVED** [review]

**Pull request merged**